app/src/main/java/com/example/whatTravel/Views/NearbySearchListAdapter.java

```diff
@@ -0,0 +1,68 @@
+package com.example.whatTravel.Views;
+
+import android.graphics.Bitmap;
+import android.support.v4.util.LruCache;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import com.android.volley.toolbox.ImageLoader;
+import com.example.whatTravel.API.WTApiLoadManager;
+import com.example.whatTravel.Models.NearbySearchResult;
+import com.example.whatTravel.R;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.example.whatTravel.Views.NearbyLocationItemView.ViewHolder;
+
+import org.w3c.dom.Text;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by jiecongwang on 2/5/15.
+ */
+public class NearbySearchListAdapter extends RecyclerView.Adapter {
+
+
+    private final Set<String> ids = Sets.newHashSet();
```

Comment:
Use Arraylist.contains()?
and   
 @Override
    public boolean equals(Object object)
    {
        boolean sameSame = false;

        if (object != null && object instanceof NearbySearchResult)
        {
            sameSame = this.id == ((NearbySearchResult) object).id;
        }
        return sameSame;
    }